keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/constraint/BakeConstraintEvaluator.kt

```diff
@@ -0,0 +1,58 @@
+package com.netflix.spinnaker.keel.bakery.constraint
+
+import com.netflix.frigga.ami.AppVersion
+import com.netflix.spinnaker.keel.api.DeliveryConfig
+import com.netflix.spinnaker.keel.api.Environment
+import com.netflix.spinnaker.keel.api.artifacts.DebianArtifact
+import com.netflix.spinnaker.keel.api.artifacts.DeliveryArtifact
+import com.netflix.spinnaker.keel.api.artifacts.VirtualMachineOptions
+import com.netflix.spinnaker.keel.bakery.api.BakeConstraint
+import com.netflix.spinnaker.keel.clouddriver.ImageService
+import com.netflix.spinnaker.keel.clouddriver.model.NamedImage
+import com.netflix.spinnaker.keel.constraints.ConstraintEvaluator
+import com.netflix.spinnaker.keel.constraints.SupportedConstraintType
+import com.netflix.spinnaker.kork.dynamicconfig.DynamicConfigService
+import kotlinx.coroutines.runBlocking
+import org.slf4j.LoggerFactory
+import org.springframework.context.ApplicationEventPublisher
+
+class BakeConstraintEvaluator(
```

Comment:
This should go away now, no? The cluster will remain in a `HAPPY` state until a new image is build by the `ImageHandler`, causing a diff for the cluster, right?

Reply:
No, because the cluster works on the artifact version not the AMI. We still need to prevent promotion until the AMI exists.